script/validate-electron-version.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable no-sync */
+/// <reference path="./globals.d.ts" />
+
+import * as distInfo from './dist-info'
+
+type ChannelToValidate = 'production' | 'beta'
+
+/**
+ * This object states the valid/expected Electron versions for each publishable
+ * channel of GitHub Desktop.
+ *
+ * The purpose of this is to ensure that we don't accidentally publish a
+ * production/beta/test build with the wrong version of Electron, which could
+ * cause really bad regressions to our users, and also the inability to go back
+ * to a previous version of GitHub Desktop without losing all settings.
+ */
+const ValidElectronVersions: Record<ChannelToValidate, string> = {
+  production: '19.0.0',
+  beta: '19.0.0',
+}
+
+const channel = getChannelToValidate()
```

Comment:
Other thoughts - But just minor nit (feel free to ignore), we could probably reuse the `distInfo.isPublishable` by doing something like and not have to make our own channel checking stuff in this file. Plus added benefit of clearing only checking for publishable builds. 

```
const expectedVersion = ValidElectronVersions[distInfo.channel];
const isExpectationDefinedForChannel = expectedVersion !== undefined
if(!distInfo.isPublishable() || !isExpectationDefinedForChannel) {
  // No need to validate console log
  process.exit(0)
}
```

Reply:
I rewrote this part several times and didn't quite like the combination of `isPushable` with additional checks 🤔 If, in the end, I needed to check the whether or not the channel is in `ValidElectronVersions`… 🤷